src/pet.js

```diff
@@ -0,0 +1,84 @@
+const AGE_INCREMENT = 1; 
+const MAXIMUM_AGE = 30;
+
+const HUNGER_INCREMENT = 5;
+const HUNGER_DECREMENT = 3;
+const MINIMUM_HUNGER = 0;
+const HUNGER_THRESHOLD = 5;
+const MAXIMUM_HUNGER = 10;
+
+const FITNESS_DECREMENT = 3;
+const FITNESS_INCREMENT = 4;
+const MAXIMUM_FITNESS = 10;
+const FITNESS_THRESHOLD = 3;
+const MINIMUM_FITNESS = 0;
+
+const DEFAULT_CONDITION = "I feel great!";
+const DEATH_STATEMENT = 'Your pet is no longer alive :(';
+
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = MINIMUM_HUNGER;
+    this.fitness = MAXIMUM_FITNESS;
+    this.children = [];
+} 
+Pet.prototype = {
+    get isAlive() {
+      return this.age < MAXIMUM_AGE && this.hunger < MAXIMUM_HUNGER && this.fitness > MINIMUM_FITNESS;
+    }
+  };
+Pet.prototype.growUp = function() {
+    if (!this.isAlive) { 
+        throw new Error(DEATH_STATEMENT);
+        }
+    this.age += AGE_INCREMENT;
+    this.hunger += HUNGER_INCREMENT;
+    this.fitness -= FITNESS_DECREMENT;
+};
+Pet.prototype.walk = function() {
+    if (!this.isAlive) { 
+        throw new Error(DEATH_STATEMENT);
+        }
+   if((this.fitness + FITNESS_INCREMENT) <= MAXIMUM_FITNESS){
+    this.fitness += FITNESS_INCREMENT;
+   } else {
+    this.fitness = MAXIMUM_FITNESS;
+   }
+};
+Pet.prototype.feed = function() {
```

Comment:
As above for `walk()`. The comparison should be done between the current value of hunger and the minimum possible, and only then do the decrement or not.

Reply:
As above